JUnit testing with server-only mods fails to locate the game

Recently, I've been trying to do some unit-testing on a mod combined with the new `serverOnlyMinecraftJar()` loom setting.
The unit-tests fail with the following error:
```
[16:17:03] [ERROR] [FabricLoader/GameProvider]: Minecraft game provider couldn't locate the game! The game may be absent from the class path, lacks some expected files, suffers from jar corruption or is of an unsupported variety/version.
```

My `build.gradle`:
```groovy
plugins {
    id 'java'
    id 'fabric-loom' version '1.2.7'
}

dependencies {
    minecraft "com.mojang:minecraft:${project.minecraft_version}"
    mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
    modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
    modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"

    testImplementation "net.fabricmc:fabric-loader-junit:${project.loader_version}"
}
loom {
    serverOnlyMinecraftJar()
}

test {
    useJUnitPlatform()
}

tasks.withType(JavaCompile).configureEach {
    it.options.getRelease().convention(17)
}

...
```
and `gradle.properties`:
```properties
minecraft_version=1.19.4
yarn_mappings=1.19.4+build.1
loader_version=0.14.18
fabric_version=0.78.0+1.19.4
```

Through some debugging, I found that [MinecraftGameProvider:193](https://github.com/FabricMC/fabric-loader/blob/912233f2102e02ddf9daf7b6ec9b29cf6dd1a8d6/minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/MinecraftGameProvider.java#L193) tries to find an origin for `MC_CLIENT`, which does not exist for server-only mods.
This is caused by [FabricLoaderLauncherSessionListener:42](https://github.com/FabricMC/fabric-loader/blob/912233f2102e02ddf9daf7b6ec9b29cf6dd1a8d6/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java#LL42C36-L42C36) from `fabric-loader-junit`, which hardcodes the environment to `EnvType.CLIENT`.

The following origins are detected in an server-only Fabric workspace:
![image](https://user-images.githubusercontent.com/35002345/236637176-0bb00ebe-ed1a-4ef6-9516-45b9d208f5be.png)


As a workarround, this could probably be tweaked using a custom LauncherSessionListener. However, I think it would be good if there was an "official" solution for this.

--- a/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
+++ b/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
@@ -17,7 +17,6 @@
 package net.fabricmc.loader.impl.junit;
 
 import java.util.Locale;
-import java.util.Optional;
 
 import org.junit.platform.launcher.LauncherSession;
 import org.junit.platform.launcher.LauncherSessionListener;

--- a/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
+++ b/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
@@ -16,6 +16,9 @@
 
 package net.fabricmc.loader.impl.junit;
 
+import java.util.Locale;
+import java.util.Optional;
+
 import org.junit.platform.launcher.LauncherSession;
 import org.junit.platform.launcher.LauncherSessionListener;
 
@@ -38,8 +41,12 @@ public class FabricLoaderLauncherSessionListener implements LauncherSessionListe
 		final Thread currentThread = Thread.currentThread();
 		final ClassLoader originalClassLoader = currentThread.getContextClassLoader();
 
+		// parse the test environment type, defaults to client
+		final EnvType envType = Optional.ofNullable(System.getProperty(SystemProperties.SIDE))
+				.map(FabricLoaderLauncherSessionListener::parseEnvType).orElse(EnvType.CLIENT);
+
 		try {
-			knot = new Knot(EnvType.CLIENT);
+			knot = new Knot(envType);
 			classLoader = knot.init(new String[]{});
 		} finally {
 			// Knot.init sets the context class loader, revert it back for now.
@@ -58,5 +65,16 @@ public class FabricLoaderLauncherSessionListener implements LauncherSessionListe
 	public void launcherSessionClosed(LauncherSession session) {
 		final Thread currentThread = Thread.currentThread();
 		currentThread.setContextClassLoader(launcherSessionClassLoader);
+	}
+
+	private static EnvType parseEnvType(String side) {
+		switch (side.toLowerCase(Locale.ROOT)) {
+		case "client":
+			return EnvType.CLIENT;
+		case "server":
+			return EnvType.SERVER;
+		default:
+			throw new RuntimeException("Invalid side provided: must be \"client\" or \"server\"!");
+		}
 	}
 }

--- a/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
+++ b/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
@@ -42,8 +42,7 @@ public class FabricLoaderLauncherSessionListener implements LauncherSessionListe
 		final ClassLoader originalClassLoader = currentThread.getContextClassLoader();
 
 		// parse the test environment type, defaults to client
-		final EnvType envType = Optional.ofNullable(System.getProperty(SystemProperties.SIDE))
-				.map(FabricLoaderLauncherSessionListener::parseEnvType).orElse(EnvType.CLIENT);
+		final EnvType envType = EnvType.valueOf(System.getProperty(SystemProperties.SIDE, EnvType.CLIENT.name()).toUpperCase(Locale.ROOT));
 
 		try {
 			knot = new Knot(envType);

--- a/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
+++ b/junit/src/main/java/net/fabricmc/loader/impl/junit/FabricLoaderLauncherSessionListener.java
@@ -64,15 +64,4 @@ public class FabricLoaderLauncherSessionListener implements LauncherSessionListe
 		final Thread currentThread = Thread.currentThread();
 		currentThread.setContextClassLoader(launcherSessionClassLoader);
 	}
-
-	private static EnvType parseEnvType(String side) {
-		switch (side.toLowerCase(Locale.ROOT)) {
-		case "client":
-			return EnvType.CLIENT;
-		case "server":
-			return EnvType.SERVER;
-		default:
-			throw new RuntimeException("Invalid side provided: must be \"client\" or \"server\"!");
-		}
-	}
 }

